src/mongo/delegates/MongoTripodTables.class.php

```diff
@@ -378,6 +414,332 @@ public function generateTableRows($tableType,$resource=null,$context=null)
+                    {
+                        return $this->rewriteVariableValue($conditionalSpec[$path], $dest);
+                    }
+
+                    return $this->getComputedValue($nestedComputedFunctions[0], $conditionalSpec[$path], $dest);
+
+                }
+                else
+                {
+                    return $this->rewriteVariableValue($conditionalSpec[$path], $dest);
+                }
+            }
+
+        }
+        return $value;
+    }
+
+    /**
+     * @param mixed $value The value to replace, if it contains a variable
+     * @param array $dest The table row document to save
+     * @param string|null $setType Force the return to be set to specified type
+     * @return mixed
+     */
+    protected function rewriteVariableValue($value, array &$dest, $setType=null)
+    {
+        if(is_string($value))
+        {
+            if(strpos($value, '$') === 0)
+            {
+                $key =  str_replace('$','', $value);
+                if(isset($dest[$key]))
+                {
+                    return $this->castValueType($dest[$key], $setType);
+                }
+                return null;
+            }
+            else
+            {
+                return $this->castValueType($value, $setType);
+            }
+        }
+        elseif(is_array($value))
+        {
+            if($this->isFunction($value))
+            {
+                $function = array_keys($value);
+                return $this->getComputedValue($function[0], $value, $dest);
+            }
+            $aryValue = array();
+            foreach($value as $v)
+            {
+                $aryValue[] = $this->rewriteVariableValue($v, $dest);
+            }
+            return $aryValue;
+        }
+
+        return $this->castValueType($value, $setType);
+    }
+
+    /**
+     * @param mixed $value
+     * @return bool
+     */
+    protected function isFunction($value)
+    {
+        return (is_array($value) && count(array_keys($value)) === 1 && count(array_intersect(array_keys($value), self::$computedFieldFunctions)) ===1);
+    }
+
+    /**
+     * @param mixed $value
+     * @param string|null $type
+     * @return mixed
+     */
+    protected function castValueType($value, $type=null)
+    {
+        switch($type)
+        {
+            case 'string':
+                $value = (string)$value;
+                break;
+            case 'bool':
+            case 'boolean':
+                $value = (bool)$value;
+                break;
+            case 'numeric':
+                if((!is_int($value)) && !is_float($value))
+                {
+                    if($value == (string)(int)$value)
+                    {
+                        $value = (int)$value;
+                    }
+                    else
+                    {
+                        $value = (float)$value;
+                    }
+                }
+                break;
+        }
+        return $value;
```

Comment:
If type is null what happens? If you're allowing type to be null, do you need something to try and work out what type the value is?